planner/core/plan_cost_ver2.go

```diff
@@ -396,6 +396,7 @@ func (p *PhysicalTopN) getPlanCostVer2(taskType property.TaskType, option *PlanC
 
 	rows := getCardinality(p.children[0], option.CostFlag)
 	n := math.Max(1, float64(p.Count+p.Offset))
+	n = math.Min(n, rows)
```

Comment:
In go1.21, we can replace it with ```min```. ```min```/```max``` is a new built-in function.

Reply:
gotit